storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -90,6 +92,21 @@ public Cluster(INimbus nimbus, Map<String, SupervisorDetails> supervisors, Map<S
         }
         this.conf = storm_conf;
     }
+
+    /**
+     * Get a copy of this cluster object
+     */
+    public Cluster getCopy() {
+        HashMap<String, SchedulerAssignmentImpl> newAssignments = new HashMap<String, SchedulerAssignmentImpl>();
+        for (Map.Entry<String, SchedulerAssignmentImpl> entry : this.assignments.entrySet()) {
+            newAssignments.put(entry.getKey(), new SchedulerAssignmentImpl(entry.getValue().getTopologyId(), entry.getValue().getExecutorToSlot()));
+        }
+        Map newConf = new HashMap<String, Object>();
+        newConf.putAll(this.conf);
```

Comment:
a lot of other existing function in the cluster use "this" to as well.  I personally like to use "this" so that I can easily tell that a variable is member variable and a function is a method of the class